test/integration/mesh_uri.cc

```diff
@@ -0,0 +1,232 @@
+unsigned char *g_imageBuffer = nullptr;
+msgs::Image g_image;
+
+/////////////////////////////////////////////////
+void cameraCb(const msgs::Image & _msg)
+{
+  ASSERT_EQ(msgs::PixelFormatType::RGB_INT8,
+      _msg.pixel_format_type());
+
+  std::lock_guard<std::mutex> lock(mutex);
+  g_image = _msg;
+  g_count++;
+}
+
+std::string meshModelStr(bool _static = false)
+{
+  // SDF string consisting of a box mesh. Note that "unit_box"
+  // is a mesh that comes pre-registered with MeshManager so
+  // it should be able to load this as a mesh.
+  // Similarly the user can create a common::Mesh object and add that
+  // to the MeshManager by calling MeshManager::Instance()->AddMesh;
+  return std::string("<?xml version=\"1.0\" ?>") +
+      "<sdf version='1.6'>" +
+      "<model name='spawned_model'>" +
+      "<link name='link'>" +
+      "<visual name='visual'>" +
+      "<geometry><mesh><uri>name://unit_box</uri></mesh></geometry>" +
+      "</visual>" +
+      "<collision name='visual'>" +
+      "<geometry><mesh><uri>name://unit_box</uri></mesh></geometry>" +
+      "</collision>" +
+      "</link>" +
+      "<static>" + std::to_string(_static) + "</static>" +
+      "</model>" +
+      "</sdf>";
+}
+
+//////////////////////////////////////////////////
+class MeshUriTest : public InternalFixture<::testing::Test>
+{
+};
+
+/////////////////////////////////////////////////
+// Test spawning mesh by name and verify that the mesh is correctly
+// spawned in both rendering and physics. Cameras should see the spawned
+// mesh and spawend object should collide with one another on the physics
+// side.
+TEST_F(MeshUriTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(SpawnMeshByName))
+{
+  // Start server
+  ServerConfig serverConfig;
+  const auto sdfFile = common::joinPaths(std::string(PROJECT_SOURCE_PATH),
+    "test", "worlds", "camera_sensor_scene_background.sdf");
+  serverConfig.SetSdfFile(sdfFile);
+
+  Server server(serverConfig);
+  EXPECT_FALSE(server.Running());
+  EXPECT_FALSE(*server.Running(0));
+
+  // Create a system just to get the ECM
+  EntityComponentManager *ecm{nullptr};
+  test::Relay testSystem;
+  testSystem.OnPreUpdate([&](const UpdateInfo &,
+                             EntityComponentManager &_ecm)
+      {
+        ecm = &_ecm;
+      });
+
+  server.AddSystem(testSystem.systemPtr);
+
+  // Run server and check we have the ECM
+  EXPECT_EQ(nullptr, ecm);
+  server.Run(true, 1, false);
+  EXPECT_NE(nullptr, ecm);
+
+  auto entityCount = ecm->EntityCount();
+
+  // Subscribe to the camera topic
+  transport::Node node;
+  g_count = 0;
+  node.Subscribe("/camera", &cameraCb);
+
+  // Run and make sure we received camera images
+  server.Run(true, 100, false);
+  for (int i = 0; i < 100 && g_count <= 0; ++i)
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  EXPECT_LT(0u, g_count);
+
+  // Empty scene - camera should see just red background
+  unsigned int bufferSize = g_image.width() *g_image.height() * 3u;
```

Comment:
We need to lock the mutex before accessing `g_image`.

Reply:
added lock. [4f381ce](https://github.com/gazebosim/gz-sim/pull/2007/commits/4f381ce88d1f7a1c8d7718ad0bc6fdfbea97cec1)